File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;

class Solution {
    
  private ArrayDeque<Integer> stack = new ArrayDeque<>();
  
  public int solution(int[] order) {
    int box =1;
    int idx = 0;

    while (box < order.length + 1) {
      stack.add(box++);
        
      while (stack.getLast() == order[idx]) {
    
        stack.pollLast();
        idx++;
          
        if (stack.isEmpty()) break;
      }
    }

    return idx;
  }
}

File: 프로그래머스/2/131704. 택배상자/택배상자.py
def solution(order):
    
    helper_c = []
    box = 1
    i = 0
    while box < len(order) + 1:
        helper_c.append(box)
        box += 1
          
        while helper_c[-1] == order[i]:
            helper_c.pop()
            i += 1
            if not helper_c: break
    
    return i